### PR TITLE
update the matrix filter to be like bf4.2

### PR DESCRIPTION
**Pull request by Quick-Flash**
* better version of matrix
* adds max Hz

### CODE REVIEW

File: src/main/sensors/gyroanalyse.c [hunk 1]
```diff
@@ -245,54 +269,91 @@ static FAST_CODE_NOINLINE void gyroDataAnalyseUpdate(gyroAnalyseState_t *state)
...
+            // get centerFreq in Hz from weighted bins
+            float centerFreq = dynNotchMaxHz;
+            float fftMeanIndex = 0;
             if (fftSum > 0) {
-                // idx was shifted by 1 to start at 1, not 0
-                fftMeanIndex = (fftWeightedSum / fftSum) - 1;
-                // the index points at the center frequency of each bin so index 0 is actually 16.125Hz
-                centerFreq = constrain(fftMeanIndex * fftResolution, dynNotchMinHz, dynNotchMaxCentreHz);
+                fftMeanIndex = (fftWeightedSum / fftSum);
+                centerFreq = fftMeanIndex * fftResolution;
```
Comment (gretel) [inline src/main/sensors/gyroanalyse.c:329]: Codacy Issue found: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by gretel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (loutwice) [reply src/main/sensors/gyroanalyse.c:329]: dont pr this to master @Quick-Flash , let your feature branch evolve and make all other update from catching betaflight code going there - the merge to master will be when all the code is ready - tested - and safe . thank you @Quick-Flash and @gretel

Comment (gretel) [reply src/main/sensors/gyroanalyse.c:329]: i tried doing QA for months. now it's a bit late to have that sort of rules, don't you think? i have left the organization and working on a fork. regards

**Comment by gretel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

**Comment by gretel** [comment]
censorships sucks 👅

**Comment by loutwice** [comment]
> censorship sucks 👅

what you mean @gretel ?

**Comment by gretel** [comment]
look above - the comment was deleted so was my reply. guess i used some "bad words" not suitable for the.. public opinion.

**Comment by loutwice** [comment]
> > censorship sucks 👅
> 
> what you mean @gretel ?

hoooo i didnt see somone deleted our discussion./tought/recommandation

**Comment by gretel** [comment]
image

**Comment by gretel** [comment]
@loutwice are you kidding me? i did delete no nothing. how about you stop blaming me and check on your emuf friends? i'm outta here.

**Comment by nerdCopter** [comment]
* i read it as loutwice asked a question (because these deletion do not make sense), not a blame.
* it might be the back-end systems like Codacy.  
* i have reset Codacy integration today, hopefully it's recent failures are fixed.

**Comment by gretel** [comment]
just get my bot off and me.

**Comment by loutwice** [comment]
> @loutwice are you kidding me? i did delete no nothing. how about you stop blaming me and check on your emuf friends? i'm outta here.

loook @gretel this is what i see from my side , that is what i don't understand :scan we discuss thing ?

image

**Comment by nerdCopter** [comment]
@Quick-Flash , would this require an MSP change?

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649&bid=18187817)

File: src/main/sensors/gyroanalyse.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:329]: Codacy found an issue: [Redundant initialization for 'centerFreq'. The initialized value is overwritten before it is read.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5547649)

**Review by nerdCopter — APPROVED** [review]
* Tests are positive.

**Pull request merged**